### PR TITLE
Export wrapper errors for use in service catalog

### PR DESCRIPTION
**Pull request by jpeeler**
Share the constants!

### CODE REVIEW

File: v2/errors.go [hunk 1]
```diff
@@ -70,11 +70,11 @@ func IsConflictError(err error) bool {
 }
 
 const (
-	asyncErrorMessage     = "AsyncRequired"
-	asyncErrorDescription = "This service plan requires client support for asynchronous service operations."
+	AsyncErrorMessage     = "AsyncRequired"
```
Comment (pmorie) [inline v2/errors.go]: I like the intent here, but I think some exported methods from the fake package that just give you canned versions of these errors is probably more useful - revise to that?

File: v2/fake/fake.go [hunk 2]
```diff
@@ -224,3 +224,19 @@ type UnbindReaction struct {
 	Response *v2.UnbindResponse
 	Error    error
 }
+
+func AsyncRequiredErrorMessage() string {
```
Comment (pmorie) [inline v2/fake/fake.go]: These should just give back the error, ie:

```go

func AsyncRequiredError() error {
  // ...
}
```

File: v2/fake/fake.go [hunk 3]
```diff
@@ -224,3 +224,19 @@ type UnbindReaction struct {
 	Response *v2.UnbindResponse
 	Error    error
 }
+
+func AsyncRequiredErrorMessage() error {
+	return errors.New(v2.AsyncErrorMessage)
```
Comment (pmorie) [inline v2/fake/fake.go]:
```go
func AsyncRequiredError error {
  return v2.HTTPStatusCodeError{
    ...
  }
}
```

File: v2/fake/fake.go [hunk 4]
```diff
@@ -224,3 +225,23 @@ type UnbindReaction struct {
 	Response *v2.UnbindResponse
 	Error    error
 }
+
+func strPtr(s string) *string {
+	return &s
+}
+
+func AsyncRequiredError() error {
```
Comment (pmorie) [inline v2/fake/fake.go]: You should add tests for these methods to ensure they correctly trip the helper functions in errors.go

**Comment by jpeeler** [comment]
I had to shuffle the wrappers out of the fake package as we had originally discussed due to import errors. But I think it works better there as the constants themselves no longer required exporting.

File: v2/errors.go [hunk 5]
```diff
@@ -116,3 +120,21 @@ func IsAppGUIDRequiredError(err error) bool {
 
 	return *statusCodeError.Description == appGUIDRequiredErrorDescription
 }
+
+// AsyncRequiredError returns error for required asynchronous operations
+func AsyncRequiredError() error {
```
Comment (pmorie) [inline v2/errors.go]: These should be in the fake package.

File: v2/fake/fake.go [hunk 6]
```diff
@@ -224,3 +225,25 @@ type UnbindReaction struct {
 	Response *v2.UnbindResponse
 	Error    error
 }
+
+func strPtr(s string) *string {
+	return &s
+}
+
+// AsyncRequiredError returns error for required asynchronous operations
```
Comment (pmorie) [inline v2/fake/fake.go]: missing period; wrap comments at 80 chars

**Comment by pmorie** [comment]
Thanks for the patch

**Pull request merged**